library/std/src/env.rs

```diff
@@ -633,11 +629,6 @@ impl Error for JoinPathsError {
 ///     None => println!("Impossible to get your home dir!"),
 /// }
 /// ```
-#[deprecated(
```

Comment:
idea: what if we leave it deprecated for a few more versions, while changing the behavior now? this makes it less likely that bugs will happen as a consequence of compiling with an older compiler than the crate assumed (but may not communicate with `rust-version`)

Reply:
Ok, I've kept it deprecated.

Reply:
We've never undeprecated a function before so this is a somewhat unique and unprecedented situation. I don't think general rules necessarily apply nor will it set a precedent for other changes.

Reply:
I don't think we should leave it deprecated in this *particular* case either.

Reply:
We discussed this in today's libs-api meeting. The conclusion we ended up coming to was that we can leave this deprecated on nightly until it ships in stable, and then undeprecate it for the *next* stable.